integrations/integration.php

```diff
@@ -135,18 +156,114 @@ public function get_slug(): string {
+		 *
+		 * @var array<string>
+		 */
+		$statuses = [];
+
+		foreach ( $this->vip_configs as $vip_config ) {
+			if ( $this->get_value_from_config( $vip_config, 'org', 'status' ) === Org_Integration_Status::BLOCKED ) {
+				return [ Org_Integration_Status::BLOCKED ];
+			}
+	
+			// Look into network_sites config before and then fallback to env config.
+			$statuses[] = $this->get_value_from_config( $vip_config, 'network_sites', 'status' ) ??
+				$this->get_value_from_config( $vip_config, 'env', 'status' );
+		}
+
+		return $statuses;
+	}
+
+	/**
+	 * Get configs of the integration in context of current site.
+	 *
+	 * @return array<VipIntegrationConfig> Returns an array of configs if the integration have multiple configs else single config object.
+	 *
+	 * @private
+	 */
+	public function get_site_configs() {
+		/**
+		 * Array containing configs of the integration.
+		 *
+		 * @var array<VipIntegrationConfig>
+		 */
+		$configs = [];
+
+		// Get configs of the integration from configurations provided by VIP.
+		foreach ( $this->vip_configs as $vip_config ) {
+			if ( is_multisite() ) {
+				$config = $this->get_value_from_config( $vip_config, 'network_sites', 'config' );
+
+				// If network site config is not found then fallback to env config if it exists.
+				if ( empty( $config ) && true === $this->get_value_from_config( $vip_config, 'env', 'cascade_config' ) ) {
+					$config = $this->get_value_from_config( $vip_config, 'env', 'config' );
+				}
+			} else {
+				$config = $this->get_value_from_config( $vip_config, 'env', 'config' );
+			}
+
+			if ( isset( $config ) ) {
+				$configs[] = $config;
+			}
+		}
+
+		// Return config object if integration have only one config else return all configs.
+		return ( ! $this->have_multiple_setups && isset( $configs[0] ) ) ? $configs[0] : $configs;
+	}
+
+	/**
+	 * Get config value based on given type and key.
+	 *
+	 * @param VipIntegrationConfig $vip_config  Configurations provided by VIP.
+	 * @param string               $config_type Type of the config whose data is needed i.e. org, env, network-sites etc.
+	 * @param string               $key Key of the config from which we have to extract the data.
+	 *
+	 * @return null|string|array Returns `null` if key is not found, `string` if key is "status" and `array` if key is "config".
+	 */
+	private function get_value_from_config( $vip_config, string $config_type, string $key ) {
+		$value = null;
+
+		if ( ! in_array( $config_type, [ 'org', 'env', 'network_sites' ], true ) ) {
+			trigger_error( 'config_type param (' . esc_html( $config_type ) . ') must be one of org, env or network_sites.', E_USER_WARNING ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_trigger_error
+		} elseif ( isset( $vip_config[ $config_type ] ) ) {
+
+			// Look for key inside org or env config.
+			if ( 'network_sites' !== $config_type && isset( $vip_config[ $config_type ][ $key ] ) ) {
+				return $vip_config[ $config_type ][ $key ];
+			}
```

Comment:
I'm not even sure if we need to concept of org configs down at this level either.

When generating configmaps, we're already dealing with only one specific environment. So we can decide what the configs are for that particular environment (based on higher levels if needed), and pass down the resulting env configs along with network site configs if it's a MS.

Reply:
Currently from code point of view we do support having config on `org` level but in actual we don't have any config from `ConfigMap` so feel free to let me know if you like me to add a condition here which throws the error if config is provided on `org` level. Personally I think it is fine as it is.